lib/msf/base/config.rb

```diff
@@ -50,6 +50,24 @@ def self.get_config_root
     end
   end
 
+  # Determines the git hash for this source tree
+  #
+  # @return [String] the git hash for this source tree
+  def self.get_git_hash
+    msfrelease = File.join(InstallRoot, 'msfrelease.txt')
+    hash = 'unknown'
+    @git_hash ||= begin
+      if File.exist?(msfrelease)
+        hash = File.read(msfrelease)
+      else
+        if File.exist?(File.join(InstallRoot, '.git'))
+          hash = `git rev-parse HEAD`
```

Comment:
I'm assuming (yes, even willing to accept the side-effects of that) that any Windows build will probably:
1. not have a .git directory (it won't have it in the omnibus installer), and/or
2. will have a msfrelease.txt  file

Apologies in advance to the legions of people doing metasploit develelopment on Windows.